File: core/src/com/spacefist/state/abst/State.java
package com.spacefist.state.abst;

/**
 * Every state machine state implements State
 */
public interface State {
    void update();

    /**
     * Called when this state becomes the active state in
     * the state machine.
     */
    void enteringState();

    /**
     * Called before another state becomes the active state.
     */
    void exitingState();
}

File: OldProject/SpaceFist/SpaceFist/State/Abstract/State.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SpaceFist.State.Abstract
{
    /// <summary>
    /// Every state machine state implements State
    /// </summary>
    public interface State
    {
        void Update();

        /// <summary>
        /// Called when this state becomes the active state in 
        /// the state machine.
        /// </summary>
        void EnteringState();

        /// <summary>
        /// Called before another state becomes the active state.
        /// </summary>
        void ExitingState();
    }
}
